File: level3/가장_긴_팰린드롬.cpp
#include <iostream>
#include <string>
//pass
using namespace std;

int longest;

int solution(string s) {

    int answer=0;
    longest = 1;

    for(int i=0; i<s.size()-1; i++){
        if(s[i] == s[i+1]){
            if(longest == 1)
                longest = 2;

            int index = 1;
            while(true){
                if(i-index < 0 || i+1+index > s.size()-1 || s[i-index] != s[i+1+index])
                    break;
                if((2*index+2) > longest)
                    longest = 2*index+2;
                    index++;
            }
        }

        int index = 1;
        while(true){
            if(i-index < 0 || i+index > s.size()-1 || s[i-index] != s[i+index])
                break;
            if((2*index+1) > longest)
                longest = 2*index+1;
                index++;
        }
    }
    return longest;
}

File: JavaScript/level3/가장_긴_팰린드롬.js
function solution(s)
{
    var answer = 1;

    for(let i=1; i<s.length-1; i++){
        let temp = 1;
        for(let j=1; j<=Math.min(s.length-1-i, i-0); j++){
            if(s[i-j] ===s[i+j])
                temp += 2;
            else
                break;
        }
        answer = Math.max(answer, temp);
    }
    
    for(let i=0; i<s.length-1; i++){
        if(s[i] === s[i+1]){
            let temp = 2;
            for(let j=1; j<=Math.min(s.length-2-i, i-0); j++){
                if(s[i-j] ===s[i+1+j])
                    temp += 2;
                else
                    break;
            }
            answer = Math.max(answer, temp);
        }
    }
    
    return answer;
}